File: app/ui/src/types/work-order.ts
export type WorkOrderStatus =
  | "created"
  | "scheduled"
  | "open"
  | "in_progress"
  | "tracking"
  | "pending"
  | "in_review"
  | "completed"
  | "void";

export type PriorityLevel = "low" | "normal" | "high" | "urgent";

export type WorkOrderType = "work_order" | "quote";

export interface City {
  id: string;
  code: string;
  name: string;
  is_active: boolean;
}

export interface CityListResponse {
  items: City[];
  total: number;
}

export interface AircraftBrief {
  id: string;
  registration_number: string;
  serial_number: string | null;
  make: string | null;
  model: string | null;
  year_built: number | null;
}

export interface CustomerBrief {
  id: string;
  name: string;
  email: string | null;
}

export interface WorkOrder {
  id: string;
  work_order_number: string;
  sequence_number: number;
  city: {
    id: string;
    code: string;
    name: string;
  };
  aircraft: AircraftBrief;
  work_order_type: WorkOrderType;
  status: WorkOrderStatus;
  status_notes: string | null;

  // Customer
  customer: CustomerBrief | null;

  // Dates
  due_date: string | null;
  created_date: string;
  completed_date: string | null;

  // Assignment
  lead_technician: string | null;
  sales_person: string | null;
  priority: PriorityLevel;

  // Audit
  created_by: string;
  updated_by: string | null;
  created_at: string;
  updated_at: string;

  // Item count
  item_count: number;
}

export interface WorkOrderListResponse {
  items: WorkOrder[];
  total: number;
  page: number;
  page_size: number;
}

export interface WorkOrderCreateInput {
  city_id: string;
  aircraft_id: string;
  created_by: string;
  work_order_type?: WorkOrderType;
  status?: WorkOrderStatus;
  status_notes?: string;
  due_date?: string;
  lead_technician?: string;
  sales_person?: string;
  priority?: PriorityLevel;
}

export interface WorkOrderUpdateInput {
  work_order_type?: WorkOrderType;
  status?: WorkOrderStatus;
  status_notes?: string;
  aircraft_id?: string;
  due_date?: string;
  lead_technician?: string;
  sales_person?: string;
  priority?: PriorityLevel;
  updated_by?: string;
}

// Status display helpers
export const WORK_ORDER_STATUS_LABELS: Record<WorkOrderStatus, string> = {
  created: "Created",
  scheduled: "Scheduled",
  open: "Open",
  in_progress: "In Progress",
  tracking: "Tracking",
  pending: "Pending",
  in_review: "In Review",
  completed: "Completed",
  void: "Void",
};

export const PRIORITY_LABELS: Record<PriorityLevel, string> = {
  low: "Low",
  normal: "Normal",
  high: "High",
  urgent: "Urgent",
};

export const WORK_ORDER_TYPE_LABELS: Record<WorkOrderType, string> = {
  work_order: "Work Order",
  quote: "Quote",
};

File: app/api/schemas/work_order.py
from pydantic import BaseModel, Field
from uuid import UUID
from datetime import datetime, date
from decimal import Decimal
from enum import Enum

from schemas.city import CityBrief


class WorkOrderStatus(str, Enum):
    CREATED = "created"
    SCHEDULED = "scheduled"
    OPEN = "open"
    IN_PROGRESS = "in_progress"
    TRACKING = "tracking"
    PENDING = "pending"
    IN_REVIEW = "in_review"
    COMPLETED = "completed"
    VOID = "void"


class PriorityLevel(str, Enum):
    LOW = "low"
    NORMAL = "normal"
    HIGH = "high"
    URGENT = "urgent"


class WorkOrderType(str, Enum):
    WORK_ORDER = "work_order"
    QUOTE = "quote"


class WorkOrderBase(BaseModel):
    """Base schema for work order fields."""

    work_order_type: WorkOrderType = WorkOrderType.WORK_ORDER
    status: WorkOrderStatus = WorkOrderStatus.CREATED
    status_notes: str | None = None

    # Customer
    customer_name: str | None = None
    customer_po_number: str | None = None

    # Dates
    due_date: date | None = None

    # Assignment
    lead_technician: str | None = None
    sales_person: str | None = None
    priority: PriorityLevel = PriorityLevel.NORMAL


class WorkOrderCreate(WorkOrderBase):
    """Schema for creating a work order."""

    city_id: UUID
    aircraft_id: UUID
    created_by: str


class WorkOrderUpdate(BaseModel):
    """Schema for updating a work order."""

    work_order_type: WorkOrderType | None = None
    status: WorkOrderStatus | None = None
    status_notes: str | None = None

    # Aircraft
    aircraft_id: UUID | None = None

    # Customer
    customer_name: str | None = None
    customer_po_number: str | None = None

    # Dates
    due_date: date | None = None

    # Assignment
    lead_technician: str | None = None
    sales_person: str | None = None
    priority: PriorityLevel | None = None

    updated_by: str | None = None


class AircraftBrief(BaseModel):
    """Brief aircraft info for work order response."""

    id: UUID
    registration_number: str
    serial_number: str | None
    make: str | None
    model: str | None
    year_built: int | None

    class Config:
        from_attributes = True


class WorkOrderResponse(BaseModel):
    """Response schema for a work order."""

    id: UUID
    work_order_number: str
    sequence_number: int
    city: CityBrief
    aircraft: AircraftBrief
    work_order_type: WorkOrderType
    status: WorkOrderStatus
    status_notes: str | None

    # Customer
    customer_name: str | None
    customer_po_number: str | None

    # Dates
    due_date: date | None
    created_date: datetime
    completed_date: datetime | None

    # Assignment
    lead_technician: str | None
    sales_person: str | None
    priority: PriorityLevel

    # Audit
    created_by: str
    updated_by: str | None
    created_at: datetime
    updated_at: datetime

    # Item count
    item_count: int = 0

    class Config:
        from_attributes = True


class WorkOrderListResponse(BaseModel):
    """Response schema for a list of work orders."""

    items: list[WorkOrderResponse]
    total: int
    page: int
    page_size: int
